scripts/dotnet.py

```diff
@@ -586,19 +586,28 @@ def get_dotnet_version(
 
     return sdk
 
+def get_dotnet_version_precise(
+        framework: str,
+        dotnet_path: Optional[str] = None,
+        sdk: Optional[str] = None) -> str:
+    sdk_path = get_sdk_path(dotnet_path)
+    sdk = get_dotnet_version_path(framework, dotnet_path,
```

Comment:
I think I would prefer if this is an if statement rather than a ternary since it's only doing something if the value is not None.

Reply:
Makes sense, updated.